measure/src/measure.rs

```diff
@@ -84,6 +84,20 @@ impl Measure {
     }
 }
 
+#[macro_export]
+macro_rules! measure {
+    ($val : expr, $name : expr, $var1 : ident, $var2 : ident) => {
+        let mut $var2 = Measure::start($name);
+        let $var1;
+        match ($val) {
+            value => {
+                $var2.stop();
+                $var1 = value;
+            }
+        }
```

Comment:
Can you try this so that the intended usage is possible? i.e. with
```rust
let (result, measure) = measure!({ /* some expression here */ }, "some name");
```

```suggestion
    ($val:expr, $name:tt) => {
        let mut measure = Measure::start($name);
        let result = $val;
        measure.stop();
        (result, measure)
```

Reply:
I initially did try to figure out how to get this syntax to work, but I ran into compilation issues:
`macro expansion ignores token `let` and any following`
I found the workaround was to pass in a $var1 and $var2

Reply:
But looks like we don't need to have a match statement to evaluate the expression.. nice! Your approach is cleaner I like it